Makefile

```diff
@@ -259,8 +263,21 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+.PHONY: codegen-download-openapi
+codegen-download-openapi:  build goreman  ## Download the OpenAPI spec from the local server.
+	@echo "Starting the backend server"
+	goreman start backend &
+	sleep 5
+	@echo "Downloading OpenAPI spec from dev server"
+	yarn --cwd ${UI_DIR} api:download
+	killall goreman || true
+
+.PHONY: codegen-ui
+codegen-ui: ## Generate the UI API files.
+	yarn --cwd ${UI_DIR}  api:generate
+
 .PHONY: build-ui
-build-ui: clean-ui
+build-ui: clean-ui codegen-ui
```

Comment:
build-ui can not depend on `codegen-ui` as `orval` would never be installed in this case